en/includes/quick-starts/mcp-auth-server.md

```diff
@@ -69,210 +66,301 @@ Install the following dependencies.
+        outputSchema: { result: z.number() }
+    },
+    async ({ a, b }) => {
+        const output = { result: a + b };
+        return {
+            content: [{ type: 'text', text: JSON.stringify(output) }],
+            structuredContent: output
+        };
+    }
+);
+
+// Register a dynamic greeting resource
+server.registerResource(
+    'greeting',
+    new ResourceTemplate('greeting://{name}', { list: undefined }),
+    {
+        title: 'Greeting Resource',
+        description: 'Dynamic greeting generator'
+    },
+    async (uri, { name }) => ({
+        contents: [
+            {
+                uri: uri.href,
+                text: `Hello, ${name}!`
+            }
+        ]
+    })
+);
+
+// Set up Express app
+const app = express();
 
-    ``` bash
-    npm start
-    ```
+// Enable CORS (add this block)
+app.use(
+    cors({
+        origin: '*', // Allow all origins for development; restrict in production (e.g., ['https://your-client-domain.com'])
+        exposedHeaders: ['Mcp-Session-Id'],
+    })
+);
+
+app.use(express.json());
+
+// Handle MCP requests at /mcp endpoint
+app.post('/mcp', async (req, res) => {
+    // Create a new transport for each request (stateless mode)
+    const transport = new StreamableHTTPServerTransport({
+        sessionIdGenerator: undefined,
+        enableJsonResponse: true
+    });
 
-=== "nnpm"
+    res.on('close', () => {
+        transport.close();
+    });
 
-    ``` bash
-    npm start
-    ```
+    await server.connect(transport);
+    await transport.handleRequest(req, res, req.body);
+});
 
-Configure **Claude Desktop** as the client application to test the MCP server.
+// Start the server
+const port = parseInt(process.env.PORT || '3000');
+app.listen(port, () => {
+    console.log(`Demo MCP Server running on http://localhost:${port}/mcp`);
+}).on('error', error => {
+    console.error('Server error:', error);
+    process.exit(1);
+});
 
-1. Open **Claude Desktop**.
-2. Click on **Claude Desktop > Settings > Developer**.
-3. Click on **Edit Config** button. This will open `claude_desktop_config.json` file location and you can open this file using an text editor.
+```
 
-Add the following configuration to `claude_desktop_config.json` file.
+## Run and test without auth
 
-```json title="claude_desktop_config.json"
-{
-    "mcpServers": {
-        "whoami-server": {
-            "command": "npx",
-            "args": [
-                "mcp-remote@latest",
-                "http://localhost:3001/sse"
-            ]
-        }
-    }
-}
+First run the dev server by running the following command.
+```bash
+npm start
 ```
 
-Use the following simple prompt to test the MCP server.
+User **MCP Inspector** as the client application to test the MCP server.
 
-```text
-Who am I?
-```
+ In a new terminal window, run the following command to launch MCP Inspector against the running MCP server.
+
+```bash
+npx @modelcontextprotocol/inspector http://localhost:3000/mcp
```

Comment:
@ayshsandu  Do we really need this 'http://localhost:3000/mcp' parameter? When I test (v0.17.2) I had to manually set the above URL in the MCP Inspector UI.